WooCommerce/Classes/Authentication/Jetpack Setup/Native Jetpack Setup/LoginJetpackSetupView.swift

```diff
@@ -0,0 +1,174 @@
+
+    /// Shows a transparent navigation bar without a bottom border.
+    private func configureNavigationBarAppearance() {
+        let appearance = UINavigationBarAppearance()
+        appearance.configureWithTransparentBackground()
+        appearance.backgroundColor = .systemBackground
+
+        navigationItem.standardAppearance = appearance
+        navigationItem.scrollEdgeAppearance = appearance
+        navigationItem.compactAppearance = appearance
+
+        let title = NSLocalizedString("Cancel", comment: "Button to dismiss the site credential login screen")
+        navigationItem.leftBarButtonItem = UIBarButtonItem(title: title, style: .plain, target: self, action: #selector(dismissView))
+    }
+
+    @objc
+    private func dismissView() {
+        dismiss(animated: true)
+    }
+}
+
+/// View to show the process of Jetpack setup during login.
+///
+struct LoginJetpackSetupView: View {
+    @ObservedObject private var viewModel: LoginJetpackSetupViewModel
+
+    /// Scale of the view based on accessibility changes
+    @ScaledMetric private var scale: CGFloat = 1.0
+
+    init(viewModel: LoginJetpackSetupViewModel) {
+        self.viewModel = viewModel
+    }
+
+    private var title: String {
+        viewModel.connectionOnly ? Localization.connectingJetpack : Localization.installingJetpack
+    }
+
+    /// Attributed string for the description text
+    private var descriptionAttributedString: NSAttributedString {
+        let font: UIFont = .body
+        let boldFont: UIFont = font.bold
+        let siteName = viewModel.siteURL.trimHTTPScheme()
+
+        let attributedString = NSMutableAttributedString(
+            string: String(format: Localization.description, siteName),
+            attributes: [.font: font,
+                         .foregroundColor: UIColor.text.withAlphaComponent(0.8)
+                        ]
+        )
+        let boldSiteAddress = NSAttributedString(string: siteName, attributes: [.font: boldFont, .foregroundColor: UIColor.text])
+        attributedString.replaceFirstOccurrence(of: siteName, with: boldSiteAddress)
+        return attributedString
+    }
+
+    var body: some View {
+        ScrollView {
+            VStack(alignment: .leading, spacing: Constants.blockVerticalPadding) {
+                JetpackInstallHeaderView()
+
+                // title and description
+                VStack(alignment: .leading, spacing: Constants.contentVerticalSpacing) {
+                    Text(title)
+                        .largeTitleStyle()
+                    AttributedText(descriptionAttributedString)
+                }
+
+                ForEach(viewModel.setupSteps) { step in
+                    HStack(spacing: Constants.stepItemHorizontalSpacing) {
+                        if step == viewModel.currentSetupStep, step != .done {
+                            ActivityIndicator(isAnimating: .constant(true), style: .medium)
+                        } else if step > viewModel.currentSetupStep {
```

Comment:
Nit: What do you think about moving this kind of check to the view model?

`step > viewModel.currentSetupStep` - `func isStepCompleted(step: JetpackInstallStep)`

The same for conditions where we adjust opacity and show/hide views.

IMO, if we do this we can easily unit test the view model. Also, I think this will make the code easier to read. 

I would love to know your thoughts about this.

Reply:
This makes sense! I updated this following your suggestion in 1832207606e7f1d5ce3af30e96ab3c63eaea0cf4.